designs/node-repair.md

```diff
@@ -0,0 +1,182 @@
+# Node Auto Repair 
+
+## Problem Statement
+
+Nodes can experience failure modes that cause degradation to the underlying hardware, filesystem, or container environment. Some of these failure modes are surfaced through the Node object (consider referencing some of them) while others are not surfaced at all (consider referencing some of these). A Diagnostic Agent such as [Node Problem Detector(NPD)](https://github.com/kubernetes/node-problem-detector) offers a way to surface these failures as additional status conditions on the node object.
```

Comment:
"consider referencing some of them" is in here twice and I was under the impression that you were going to add examples rather than leaving the boilerplate in here

Reply:
I missed this when going back and forth between the doc and this PR. Updated.